doc/source/fault-tolerance.rst

```diff
@@ -88,3 +88,87 @@ You can experiment with this behavior by running the following code.
             print(counter)
         except ray.exceptions.RayActorError:
             print('FAILURE')
+
+Actor Checkpointing
+~~~~~~~~~~~~~~~~~~~
+
+It's often important to be able to checkpoint the state of an actor in order to
+recover that state in the event of a failure. This can be done by having the
+actor inherit from the ``ray.actor.Checkpointable`` interface.
+
+Several things are left to the application and are not done by Ray. The reason
+for this is that applications will want to handle checkpointing in many
+different ways.
+- When to create checkpoints. The ``should_checkpoint`` method should determine
+  this.
+- What to checkpoint, how to serialize it, and where to store the checkpoint.
+  The ``save_checkpoint`` method handles this.
+- How to restore the actor state from a given checkpoint. The
+  ``load_checkpoint`` method takes care of this.
+
+Here is an example in which we checkpoint the actor state every other iteration.
+In this example, we save the checkpoints in a local file. To make this work in
+the multi-node setting, we would need to write to a distributed file system or
+some globally accessible storage.
```

Comment:
Global accessible storage or distributed FS seem like requirements for this checkpointing mechanism to work on spot instances. Maybe make this much clearer?

Reply:
It's not just spot instances, it's any multi-node setting because there is no guarantee that the new actor will be recreated on the same machine.